tag_batch.go

```diff
@@ -12,6 +12,7 @@ import (
 )
 
 // compiled regular expression - for tags - custom dimensions can have whatever they want
+// nolint:unused
```

Comment:
I don't think you need to do this now.  But this whole init section could be cleaned up.

1. As you noted `_validTagValueRegexp` is not used.  So we could remove it.  @wblakecaldwell does this regex have a purpose?
2. Since the regex is a static string, we can also use `regexp.MustCompile()` which does not return an error and will panic if the regex does not compile.  But this is exactly what `initTagBatch()` is doing so we can simplify the declaration to ```var validTagValueRegexp = regexp.MustCompile(`^[a-zA-Z0-9_-]+$`)``` and remove `initTagBatch()` completely.
3. Each Go file may have its own `init()` function [[1](https://go.dev/doc/effective_go#init)].  I don't think we need init.go to call a bunch of other init functions.  Unless there's some dependency and the order is important.  But in this case I don't think it does.

[1] https://go.dev/doc/effective_go#init

I would prefer if we could clean up the unused code but I don't want to creep the scope of this linting PR.  The `//nolint` seems fine here until we can clarify the purpose of that regex with @wblakecaldwell

Reply:
Yeah, kill that unused code till it dies from it 👍